### PR TITLE
Improved test coverage for workloadrebalancer controller

### PR DESCRIPTION
**Pull request by anujagrawal699**
**Description:**
This PR adds a new test case for cluster-wide resources and refactors the existing test structure for the WorkloadRebalancer controller. These changes aim to improve code coverage and reduce cyclomatic complexity.

**Changes:**
1. Added a new test case for cluster-wide resource rebalancing:
   - This test case ensures that the controller correctly handles cluster-wide resources, improving our coverage of different resource types.

2. Refactored the TestRebalancerController_Reconcile function:
   - Extracted the test execution logic into a separate function (runRebalancerTest).
   - Moved binding check logic into separate functions (checkBindings, checkClusterBinding, checkResourceBinding).
   - These changes reduce the cyclomatic complexity of the main test function, making it easier to understand and maintain.

**Rationale:**
- The new test case for cluster-wide resources fills a gap in our test coverage, ensuring that this important scenario is properly tested.
- The refactoring addresses a high cyclomatic complexity warning from _golangci-lint_, improving code quality and maintainability.

**Test Coverage:**
pkg/controllers/workloadrebalancer : 63.01% to 75,.4%

**What type of PR is this?**
/kind failing-test
/kind feature

**Which issue(s) this PR fixes**:
Fixes a part of #5470 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/workloadrebalancer/workloadrebalancer_controller_test.go [hunk 1]
```diff
@@ -250,51 +263,121 @@ func TestRebalancerController_Reconcile(t *testing.T) {
...
+				NamespacedName: types.NamespacedName{Name: clusterRebalancer.Name},
+			},
+			existObjects:        []client.Object{clusterDeploy, clusterBinding, clusterRebalancer},
+			existObjsWithStatus: []client.Object{clusterRebalancer},
+			wantStatus: appsv1alpha1.WorkloadRebalancerStatus{
+				ObservedWorkloads: []appsv1alpha1.ObservedWorkload{
+					{
+						Workload: clusterDeployObj,
+						Result:   appsv1alpha1.RebalanceSuccessful,
+					},
```
[the remark this answers is not in the document]
Comment (anujagrawal699) [reply pkg/controllers/workloadrebalancer/workloadrebalancer_controller_test.go:278]: Thanks i'll work on it.

**Comment by anujagrawal699** [comment]
@chaosi-zju Please take a look.

**Comment by anujagrawal699** [comment]
@chaosi-zju PTAL.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/approve

**Pull request merged**